src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -27,6 +32,43 @@ public void constructor() {
         assertEquals(new UserPrefs(), modelManager.getUserPrefs());
         assertEquals(new GuiSettings(), modelManager.getGuiSettings());
         assertEquals(new AddressBook(), new AddressBook(modelManager.getAddressBook()));
+        assertEquals(FXCollections.observableArrayList(), modelManager.getHistoryCommandList());
+    }
+    @Test
+    public void testGetHistoryCommandList() {
+        Command command = new ClearCommand();
+        String commandText = "clear command";
+
+        modelManager.setCommandHistoryText(commandText);
+        modelManager.addHistoryCommand(command);
+
+        assertEquals(1, modelManager.getHistoryCommandList().size());
+        assertEquals(commandText, modelManager.getHistoryCommandList().get(0).getOriginalCommandText());
+        assertEquals(command, modelManager.getHistoryCommandList().get(0).getCommand());
+    }
+    @Test
+    public void testSetCommandHistoryText() {
+        ModelManager model = new ModelManager();
+        String commandText = "sample command text";
+        modelManager.setCommandHistoryText(commandText);
+        Command command = new ClearCommand();
+        model.addHistoryCommand(command);
+        assertEquals("sample command text", model.getHistoryCommandList().get(0).getOriginalCommandText());
+    }
+
+    @Test
+    public void addAndGetHistoryCommand() {
+        ModelManager model = new ModelManager();
+        String commandText = "sample command text";
+        model.setCommandHistoryText(commandText);
+        Command command = new ClearCommand();
+        model.addHistoryCommand(command);
+
+        ObservableList<HistoryCommand> historyCommands = model.getHistoryCommandList();
+
+        // Check that the list contains one command and that it matches the added command
+        assertEquals(1, historyCommands.size());
+        HistoryCommand historyCommand = historyCommands.get(0);
```

Comment:
did you want to check if the contents of `historyCommand` matches what you expect

Reply:
yaya cause the constructor is being set to private for History Command